src/private/indoor_map_floor_outline_information_module.js

```diff
@@ -0,0 +1,117 @@
+var MapModule = require("./map_module");
+
+function IndoorMapFloorOutlineInformationModuleImpl(emscriptenApi) {
+
+    var _emscriptenApi = emscriptenApi;
+    var _nativeIdToIndoorMapFloorOutlineInformation = {};
+    var _callbackInvokedBeforeAssignement = {};
+    var _pendingIndoorFloorOutlineInformation = [];
+    var _ready = false;
+    var _notifyIndoorMapFloorOutlineInformationLoadedCallback = null;
+
+    var _createPendingIndoorMapFloorOutlineInformations = function() {
+        _pendingIndoorFloorOutlineInformation.forEach(function(indoorMapFloorOutlineInformation) {
+            _createAndAdd(indoorMapFloorOutlineInformation);
+        });
+
+        _pendingIndoorFloorOutlineInformation = [];
+    };
+
+    var _createAndAdd = function(indoorMapFloorOutlineInformation) {
+        var nativeId = _emscriptenApi.indoorMapFloorOutlineInformationApi.createIndoorMapFloorOutlineInformation(indoorMapFloorOutlineInformation);
+        _nativeIdToIndoorMapFloorOutlineInformation[nativeId] = indoorMapFloorOutlineInformation;
+        indoorMapFloorOutlineInformation._setNativeHandle(nativeId);
+
+        if(nativeId in _callbackInvokedBeforeAssignement){
+            delete _callbackInvokedBeforeAssignement[nativeId];
+            _fetchIndoorMapFloorOutlineInformation(nativeId);
+        }
+
+        return nativeId;
+    };
+
+    this.addIndoorMapFloorOutlineInformation = function(indoorMapFloorOutlineInformation) {
+        if (_ready) {
+            _createAndAdd(indoorMapFloorOutlineInformation);
+        }
+        else {
+            _pendingIndoorFloorOutlineInformation.push(indoorMapFloorOutlineInformation);
+        }
+    };
+    
+    this.removeIndoorMapFloorOutlineInformation = function(indoorMapFloorOutlineInformation) {
+
+        if (!_ready) {
+            var index = _pendingIndoorFloorOutlineInformation.indexOf(indoorMapFloorOutlineInformation);
+            if (index > -1) {
+                _pendingIndoorFloorOutlineInformation.splice(index, 1);
+            }
+            return;
+        }
+
+        var nativeId = indoorMapFloorOutlineInformation.getNativeId();
+        if (nativeId === undefined) {
```

Comment:
do we use 'undefined' as invalid handle elsewhere, or just value 0?

Reply:
Using undefined for nativeIds in most other apis such as building outlines, heatmap, polygon, polyline etc.